cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -158,7 +158,7 @@ func (o *objectGraph) ownerToVirtualNode(owner metav1.OwnerReference, namespace
 		tenantClusters: make(map[*node]empty),
 		tenantCRSs:     make(map[*node]empty),
 		virtual:        true,
-		forceMove:      o.getForceMove(owner.Kind, owner.APIVersion),
+		forceMove:      o.getForceMove(owner.Name, namespace, owner.Kind, owner.APIVersion, make(map[string]string)),
```

Comment:
In order to compensate the lack of labels when adding a virtual node, it is required to re-compute the `forceMove` flag when the real node is processed (rif. `objToNode`, after `existingNode.markObserved()`)

Without this, there is the risk that, forceMove will report false negatives depending on the discovery order

Reply:
ok ,added it